File: android/word/src/main/java/com/dreampany/word/ui/model/UiTask.java
/*
package com.dreampany.word.ui.model;

import android.os.Parcel;
import androidx.annotation.NonNull;

import com.dreampany.frame.data.model.BaseParcel;
import com.dreampany.frame.data.model.Task;
import com.dreampany.word.ui.enums.UiSubtype;
import com.dreampany.word.ui.enums.UiType;

*/
/**
 * Created by Hawladar Roman on 30/4/18.
 * Dreampany
 * dreampanymail@gmail.com
 *//*

public class UiTask<T extends BaseParcel> extends Task<T> {

    protected UiType type;
    protected UiSubtype subtype;
    private boolean fullscreen;
    private boolean full;


    public UiTask() {
    }

    public UiTask(boolean fullscreen) {
        this.fullscreen = fullscreen;
    }

    private UiTask(Parcel in) {
        super(in);
        type = UiType.valueOf(in.readInt());
        subtype = UiSubtype.valueOf(in.readInt());
        fullscreen = in.readByte() != 0;
        full = in.readByte() != 0;
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeInt(type == null ? -1 : type.ordinal());
        dest.writeInt(subtype == null ? -1 : subtype.ordinal());
        dest.writeByte((byte) (fullscreen ? 1 : 0));
        dest.writeByte((byte) (full ? 1 : 0));
    }

    public static final Creator<UiTask> CREATOR = new Creator<UiTask>() {
        @Override
        public UiTask createFromParcel(Parcel in) {
            return new UiTask(in);
        }

        @Override
        public UiTask[] newArray(int size) {
            return new UiTask[size];
        }
    };

    public void setUiType(UiType type) {
        this.type = type;
    }

    public void setSubtype(UiSubtype subtype) {
        this.subtype = subtype;
    }

    public void setFullscreen(boolean fullscreen) {
        this.fullscreen = fullscreen;
    }

    public void setFull(boolean full) {
        this.full = full;
    }

    public UiType getType() {
        return type;
    }

    public UiSubtype getSubtype() {
        return subtype;
    }

    public boolean isFullscreen() {
        return fullscreen;
    }

    public boolean isFull() {
        return full;
    }
}
*/

File: android/cast/src/main/kotlin/com/dreampany/cast/ui/model/UiTask.kt
package com.dreampany.cast.ui.model

import com.dreampany.cast.ui.enums.UiSubtype
import com.dreampany.cast.ui.enums.UiType
import com.dreampany.frame.data.model.BaseKt
import com.dreampany.frame.data.model.Task
import kotlinx.android.parcel.Parcelize

/**
 * Created by roman on 2019-07-13
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Parcelize
data class UiTask<T : BaseKt>(
    val fullscreen: Boolean,
    val type: UiType,
    val subtype: UiSubtype
) : Task<T>() {

/*    var type: UiType? = null
    var subtype: UiSubtype? = null*/

/*    private constructor(`in`: Parcel) : super(`in`) {
        fullscreen = `in`.readByte().toInt() != 0
        type = UiType.valueOf(`in`.readInt())
        subtype = UiSubtype.valueOf(`in`.readInt())
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        dest.writeByte((if (fullscreen) 1 else 0).toByte())
        dest.writeInt(if (type == null) -1 else type!!.ordinal)
        dest.writeInt(if (subtype == null) -1 else subtype!!.ordinal)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<UiTask<*>> = object : Parcelable.Creator<UiTask<*>> {

            override fun createFromParcel(`in`: Parcel): UiTask<*> {
                return UiTask<BaseParcelKt>(`in`)
            }

            override fun newArray(size: Int): Array<UiTask<*>?> {
                return arrayOfNulls(size)
            }
        }
    }*/
}